### ISSUE TITLE
Windows applications crash after calling storage.read

### ISSUE THREAD

**Issue by 0xNF** [mogol/flutter_secure_storage#277]
I'm currently using `flutter_secure_storage: ^5.0.0-beta.2`  and have a block of code approximately like so:

```dart
Future<String?> getRead() async {
  // Create storage
  FlutterSecureStorage storage = const FlutterSecureStorage();
  // Read value
  try {
    String? value = await storage.read(key: 'STORAGE_KEY');
    return value;
  } catch (e) {
    return null;
  }
}
```

After hitting the `read` line, the application abruptly closes without any kind of warning.

Earlier, I was at least able to get a PlatformException that said something like 'No implementation of Read for platform Windows`, but I wasn't able to catch the exact error and have been unable to get it to throw again. All I get are total crashes. 

Here's Flutter Doctor, in case you want it:

```txt
[√] Flutter (Channel beta, 2.4.0-4.2.pre, on Microsoft Windows [Version 10.0.19043.1110], locale en-US)
    • Flutter version 2.4.0-4.2.pre at C:\Users\user\Documents\projects\external\flutter   
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f18b9281c2 (2 weeks ago), 2021-07-22 14:08:30 -0700
    • Engine revision 844c29f42a
    • Dart version 2.14.0 (build 2.14.0-301.2.beta)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at C:\Users\user\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Professional 2019 16.10.1)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional
    • Visual Studio Professional 2019 version 16.10.31402.337
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.59.0)
    • VS Code at C:\Users\user\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.25.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1110]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 92.0.4515.131
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 92.0.902.62

• No issues found!
```

**Comment by Billaids** [comment]
Hey, I've had the same issue when I tried to read in a value that doesn't even exist yet.  The Windows app crashes when you try to read a value that doesn't exist. You should save the value before you read it in. @0xNF @juliansteenbakker

**Comment by juliansteenbakker** [comment]
@kennethnym Is this something you can look at?

**Comment by 0xNF** [comment]
Thanks for the responses. I tried to check the key's existence before reading, and now I'm seeing

`No implementation found for method containsKey on channel plugins.it_nomads.com/flutter_secure_storage`

```dart
Future<String?> getRead() {
    FlutterSecureStorage storage = const FlutterSecureStorage();
    bool hasKey = await storage.containsKey(key: 'STORAGE_KEY');
    String? value;
    if (hasKey) {
      value = await storage.read(key: 'STORAGE_KEY');
    }
    return value;
}
```

Do I need to manually include the `flutter_secure_storage_windows` plugin in my pubspec.yaml?

**Comment by Billaids** [comment]
@0xNF you only need to add  "flutter_secure_storage: ^5.0.0-beta.2" and it will work!

**Comment by 0xNF** [comment]
Hm, but I'm not using the unimplemented methods. the README states that readAll() and deleteAll() aren't in yet, but I'm using just read() and constainsKey().

```dart
storage.write(key: 'STORAGE KEY', value: null);

```

Also, I tried to set a default value on first launch to `null`, and it ended up throwing a Visual C++ Redistributable exception:



![image](https://user-images.githubusercontent.com/1915007/128825106-5215d879-ef1f-49f1-a9c5-4b040b2a6adc.png)

**Comment by juliansteenbakker** [comment]
Please note that #274(Windows implementations of containsKey, readAll and deleteAll) has not been released yet, but is present in develop branch.

**Comment by Billaids** [comment]
@0xNF I've had the same error message yesterday evening. I deleted with .delete() the saved keys and run flutter clean (and flutter pub get again). After these steps, you have to save your key before you read it in.

**Comment by 0xNF** [comment]
ah, `containsKey` isn't yet implemented yet. I missed that, sorry. Guess that explains that part of the error.

**Comment by juliansteenbakker** [comment]
I have added a section in the readme.md about platform implementation of functions which may clear things up. I will release a new beta later today.

**Comment by kennethnym** [comment]
Note that in the new implementation, if `null` is passed to `write`, it performs a `delete` instead, in accordance with the docs.

**Comment by Billaids** [comment]
Still waiting for the new beta

**Comment by juliansteenbakker** [comment]
> Still waiting for the new beta

Just released v5.0.0-beta.3

**Comment by 0xNF** [comment]
I can confirm with beta3 that `constainsKey` works properly and that `read` no longer crashes the application even with an undefined key.

Thanks to @kennethnym  and @juliansteenbakker for their work on this
